castep_outputs/parsers/castep_file_parser.py

```diff
@@ -967,7 +967,30 @@ def parse_castep_file(castep_file_in: TextIO,
             if Filters.MD not in to_parse:
                 continue
 
-            logger("Found MD Block (step %d)", len(curr_run["md"])+1)
+            logger("Found MD Block (step %d)", len(curr_run["md"]))
+
+            # Avoid infinite recursion
+            next(block)
+            data = parse_castep_file(block)[0]
```

Comment:
How about

```python
data, *rest = parse_castep_file(block)
assert rest == []
```

Reply:
Hm. The alternative would be:
```suggestion
            data, = parse_castep_file(block)
```
Which would also do that (though the comma being significant is not obvious and `ruff` would demand:
```suggestion
            (data,) = parse_castep_file(block)
```
Not sure I actually like the look of any of them or think they add clarity. `parse_castep_file` should, in principle, handle any unexpected structures which might result in multiple elements in the list (and error if the calculation ended in a partial state).